### PR TITLE
rauc-installer: refactoring and bug fixes

### PR DESCRIPTION
**Pull request by Bastian-Krause**
This is the fourth of a series of PRs to refactor and clean up the code base, get rid of bugs (memory leaks, use-after-frees, race conditions), simplify functions and document them.

In this iteration I focused on the `rauc-installer` module.

### CODE REVIEW

File: src/rauc-installer.c [hunk 1]
```diff
@@ -134,14 +134,20 @@ static struct install_context *install_context_new(void)
...
+
+        loop_context = g_main_loop_get_context(context->mainloop);
+
         g_free(context->bundle);
         g_mutex_clear(&context->status_mutex);
         g_main_context_unref(context->loop_context);
         g_assert_cmpint(context->status_result, >=, 0);
         g_assert_true(g_queue_is_empty(&context->status_messages));
         g_main_loop_unref(context->mainloop);
+        g_main_context_unref(loop_context);
```
Comment (ejoerns) [inline src/rauc-installer.c:150]: isn't this the same context we freed in line 146 with `g_main_context_unref(context->loop_context);` already? Or did I miss something?

The creation says:

``` c
GMainContext *loop_context = g_main_context_new();
[...]
context->mainloop = g_main_loop_new(loop_context, FALSE);
context->loop_context = loop_context;
```

File: src/rauc-installer.c [hunk 2]
```diff
@@ -163,35 +163,38 @@ static gpointer install_loop_thread(gpointer data)
         GBusType bus_type = (!g_strcmp0(g_getenv("DBUS_STARTER_BUS_TYPE"), "session"))
                             ? G_BUS_TYPE_SESSION : G_BUS_TYPE_SYSTEM;
         RInstaller *r_installer_proxy = NULL;
-        GError *error = NULL;
-        struct install_context *context = data;
+        g_autoptr(GError) error = NULL;
+        struct install_context *context = NULL;
+
+        g_return_val_if_fail(data, NULL);
```
Comment (ejoerns) [inline src/rauc-installer.c:169]: nit: in the other commits you simply did `g_return_val_if_fail(context)` which is inconsistent with this one.

**Review by ejoerns — CHANGES_REQUESTED** [review]

File: src/rauc-installer.c [hunk 3]
```diff
@@ -57,19 +60,32 @@ static void on_installer_status(GDBusProxy *proxy, GVariant *changed,
...
-                        g_queue_push_tail(&context->status_messages, g_strdup(message));
-                } else if (g_variant_lookup(changed, "Progress", "(isi)", &percentage, &message, &depth)) {
-                        g_queue_push_tail(&context->status_messages, g_strdup_printf("%3"G_GINT32_FORMAT "%% %s", percentage, message));
-                } else if (g_variant_lookup(changed, "LastError", "s", &message) && message[0] != '\0') {
-                        g_queue_push_tail(&context->status_messages, g_strdup_printf("LastError: %s", message));
+                        g_queue_push_tail(&context->status_messages,
+                                          g_steal_pointer(&message));
+                } else if (g_variant_lookup(changed, "Progress", "(isi)",
+                                            &percentage, &message, NULL)) {
+                        gchar *progress_msg = g_strdup_printf(
```
[the remark this answers is not in the document]
Comment (Bastian-Krause) [reply src/rauc-installer.c:71]: You're right, the temporary variable is not needed here. I'll remove it.

File: src/rauc-installer.c (hunk 1 above)
Comment (Bastian-Krause) [reply src/rauc-installer.c:150]: You're right, I somehow missed that. Will fix it.

**Comment by Bastian-Krause** [comment]
Force-pushed:
- wrap lines > 100 characters
- remove accidental double loop context unref in progress_msg()
- remove unneeded temporary variables

**Comment by Bastian-Krause** [comment]
Added to commit "rauc-installer: refactor install_loop_thread()":
- do not call g_signal_handlers_disconnect_by_data() with NULL pointer instance if proxy creation failed (goto later notify_complete label instead)

**Review by ejoerns — APPROVED** [review]

**Pull request merged**